mesonbuild/dependencies/pkgconfig.py

```diff
@@ -423,18 +423,20 @@ def check_pkgconfig(self, pkgbin: ExternalProgram) -> T.Optional[str]:
         if not pkgbin.found():
             mlog.log(f'Did not find pkg-config by name {pkgbin.name!r}')
             return None
+        command_as_string = ' '.join(pkgbin.get_command())
+        if r'Strawberry\perl\bin' in pkgbin.get_path():
```

Comment:
I'm slightly worried that this check is too strict. Like get_path() could return POSIX path, or it could be in another subdir. I would rather go with `'strawberry' in pkgbin.get_path().lower()`. But in any case, yes we should black list that crap, and probably do that for any binary such as gcc.exe, etc. I just now realise it's the cause of msvc env not being activated by default on GitHub Windows CI, meson find strawberry gcc...

Reply:
```python
if seems_to_be_strawberry and self.name != 'perl':
    return [None]
```

*dies*

Because we do of course want to make sure that actual perl.exe is detected, people might want to actually use that. As a script interpreter at the very least.